src/topology/path_connected.lean

```diff
@@ -0,0 +1,589 @@
+`h.extend : ℝ → X`, and `h.map : I → F` taking values in the subtype `F`,
+and also `h.extend_map : ℝ → F`.
+
+-/
+
+noncomputable theory
+open_locale classical topological_space filter
+open filter set
+
+variables {X : Type*} [topological_space X] {x y z : X} {ι : Type*}
+
+local notation `I` := Icc (0 : ℝ) 1
+
+lemma Icc_zero_one_symm {t : ℝ} : t ∈ I ↔ 1 - t ∈ I :=
+begin
+  rw [mem_Icc, mem_Icc],
+  split ; intro ; split ; linarith
+end
+
+instance I_has_zero : has_zero I := ⟨⟨0, by split ; norm_num⟩⟩
+
+@[simp, norm_cast] lemma coe_I_zero : ((0 : I) : ℝ) = 0 := rfl
+
+instance I_has_one : has_one I := ⟨⟨1, by split ; norm_num⟩⟩
+
+@[simp, norm_cast] lemma coe_I_one : ((1 : I) : ℝ) = 1 := rfl
+
+/-- Unit interval central symmetry. -/
+def I_symm : I → I := λ t, ⟨1 - t.val, Icc_zero_one_symm.mp t.property⟩
+
+local notation `σ` := I_symm
+
+@[simp] lemma I_symm_zero : σ 0 = 1 :=
+subtype.ext $ by simp [I_symm]
+
+@[simp] lemma I_symm_one : σ 1 = 0 :=
+subtype.ext $ by simp [I_symm]
+
+@[continuity]
+lemma continuous_I_symm : continuous σ :=
+by continuity!
+
+/-- Projection of `ℝ` onto its unit interval. -/
+def proj_I : ℝ → I :=
+λ t, if h : t ≤ 0 then ⟨0, left_mem_Icc.mpr zero_le_one⟩ else
+     if h' : t ≤ 1 then ⟨t, ⟨le_of_lt $ not_le.mp h, h'⟩⟩ else ⟨1, right_mem_Icc.mpr zero_le_one⟩
+
+lemma proj_I_I {t : ℝ} (h : t ∈ I) : proj_I t = ⟨t, h⟩ :=
+begin
+  unfold proj_I,
+  rw mem_Icc at h,
+  split_ifs,
+  { simp [show t = 0, by linarith] },
+  { refl },
+  { exfalso, linarith }
+end
+
+lemma range_proj_I : range proj_I = univ :=
+begin
+  rw eq_univ_iff_forall,
+  rintro ⟨t, t_in⟩,
+  use [t, proj_I_I t_in],
+end
+
+@[continuity]
+lemma continuous_proj_I : continuous proj_I :=
+begin
+  refine continuous_induced_rng' (coe : I → ℝ) rfl _,
+  have : continuous (λ t : ℝ, if t ≤ 0 then 0 else if t ≤ 1 then t else 1),
+  { refine continuous_if _ continuous_const (continuous_if _ continuous_id continuous_const) ;
+    simp [Iic_def, zero_le_one] },
+  convert this,
+  ext,
+  dsimp [proj_I],
+  split_ifs ; refl
+end
+
+variables {β : Type*}
+
+/-- Extension of a function defined on the unit interval to `ℝ`, by precomposing with
+the projection. -/
+def I_extend {β : Type*} (f : I → β) : ℝ → β :=
+f ∘ proj_I
+
+@[continuity]
+lemma continuous.I_extend {f : I → X} (hf : continuous f) : continuous (I_extend f) :=
+hf.comp continuous_proj_I
+
+lemma I_extend_extends (f : I → β) {t : ℝ} (ht : t ∈ I) : I_extend f t = f ⟨t, ht⟩ :=
+by simp [I_extend, proj_I_I, ht]
+
+@[simp] lemma I_extend_zero (f : I → β) : I_extend f 0 = f 0 :=
+I_extend_extends _ _
+
+@[simp] lemma I_extend_one (f : I → β) : I_extend f 1 = f 1 :=
+I_extend_extends _ _
+
+@[simp] lemma I_extend_range (f : I → β) : range (I_extend f) = range f :=
+begin
+  rw [I_extend, range_comp],
+  convert image_univ,
+  exact range_proj_I
+end
+
+instance : connected_space I := subtype.connected_space ⟨⟨0, by split ; norm_num⟩, is_preconnected_Icc⟩
```

Comment:
We have [set.nonempty_Icc](https://leanprover-community.github.io/mathlib_docs/data/set/intervals/basic.html#set.nonempty_Icc)